airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSourceOperations.java

```diff
@@ -61,19 +94,25 @@ public JsonSchemaType getJsonType(final JDBCType jdbcType) {
   }
 
   /**
-   * The only difference between this method and the one in {@link JdbcSourceOperations} is that
-   * the TIMESTAMP_WITH_TIMEZONE columns are also converted using the putTimestamp method.
-   * This is necessary after the JDBC upgrade from 3.13.9 to 3.13.22. This change may need to be
-   * added to {@link JdbcSourceOperations#setJsonField} in the future.
+   * The only difference between this method and the one in {@link JdbcSourceOperations} is that the
+   * TIMESTAMP_WITH_TIMEZONE columns are also converted using the putTimestamp method. This is
+   * necessary after the JDBC upgrade from 3.13.9 to 3.13.22. This change may need to be added to
+   * {@link JdbcSourceOperations#setJsonField} in the future.
    * <p/>
    * See issue: https://github.com/airbytehq/airbyte/issues/16838.
    */
   @Override
   public void setJsonField(final ResultSet resultSet, final int colIndex, final ObjectNode json) throws SQLException {
     final int columnTypeInt = resultSet.getMetaData().getColumnType(colIndex);
     final String columnName = resultSet.getMetaData().getColumnName(colIndex);
-    final JDBCType columnType = safeGetJdbcType(columnTypeInt);
+    final String columnTypeName = resultSet.getMetaData().getColumnTypeName(colIndex).toLowerCase();
 
+    final JDBCType columnType = safeGetJdbcType(columnTypeInt);
+    // TIMESTAMPLTZ data type detected as JDBCType.TIMESTAMP which is not correct
+    if ("TIMESTAMPLTZ".equalsIgnoreCase(columnTypeName)) {
+      putTimestampWithTimezone(json, columnName, resultSet, colIndex);
```

Comment:
my understanding that `ltz` types use the local timezone - do we need special handling for this? (or is the idea that the connector container uses UTC, so it works correctly?)

Reply:
> my understanding that `ltz` types use the local timezone - do we need special handling for this? (or is the idea that the connector container uses UTC, so it works correctly?)

currently in master branch TIMESTAMPLTZ datatype converts value into UTC format so i leave it as is without changes